### ISSUE TITLE
[Bug] Crash when opening 3D view with a 4K display

### ISSUE THREAD

**Issue by lopmar** [spectralcode/OCTproZ#22]
**Describe the bug**
While use a 4K display and resolution (3840 x 2160) the software crashes when opening the 3D view window.
Running the program in Qt 5.12.12 Creator generates the following Application output:

`QWindowsWindow::setGeometry: Unable to set geometry 3838x2034+0+46 (frame: 3864x2105-13-12) on QWidgetWindow/"OCTproZWindow" on "\\.\DISPLAY1". Resulting geometry: 3840x2035+0+45 (frame: 3866x2106-13-13) margins: 13, 58, 13, 13 minimum size: 1238x890 MINMAXINFO maxSize=0,0 maxpos=0,0 mintrack=2502,1851 maxtrack=0,0)
QWindowsWindow::setGeometry: Unable to set geometry 3840x2036+0+46 (frame: 3866x2107-13-12) on QWidgetWindow/"OCTproZWindow" on "\\.\DISPLAY1". Resulting geometry: 3840x2035+0+45 (frame: 3866x2106-13-13) margins: 13, 58, 13, 13 minimum size: 1238x890 MINMAXINFO maxSize=0,0 maxpos=0,0 mintrack=2502,1851 maxtrack=0,0)
Main Thread ID start emit:  0x13e0`

**To Reproduce**
Using a 4K monitor, open the 3D view of the program. It is not necessary to load any data for the crash to happen.

**Used hardware**

    OS: Win 10 64bit
    GPU: NVIDIA GeForce RTX A4000 (CUDA 11.7)
    Screens: AOC U2879VF

**Additional context**
This problem has been identified in multiple hardware configurations, always using a 4K monitor. Using the same monitor at a lower resolution (1920x1080) works fine.

**Comment by spectralcode** [comment]
Thanks for the bug report!

I was able to reproduce the bug, but only after changing the scale value in Windows 10's display settings. For some scale values (100%, 175%, 225%) a crash never occurred after opening the 3D view and for other scale values (125%, 150%, 200%, 225%) it always crashed.

In my setup (Qt 5.11.1, GTX 1080, Windows 10, 4k display connected via HDMI) the output window in QtCreator did not show the debug message you copied in your bug report (but the debug message was displayed when no crash occurred, so the message is probably not related to this issue). Instead, an error window was displayed indicating a read access violation. It looks like the crash is happening in the nvoglv64.dll which is part of the Nvidia GPU driver.

I will investigate this further, but it may take some time. Please feel free to provide any additional info if you think it may help.

Can you change the scale value in the Windows display settings as workaround?

**Comment by lopmar** [comment]
Indeed, I can replicate the problem and it only crashes for a smaller set of scaling factors in my case (125% and 200%).

Now I can also see the same exception related to nvolgv64.dll when the crash occurs (the debugger was not configured properly before).

I will use a scaling factor that works in the meantime. Thank you very much!

**Comment by spectralcode** [comment]
The bug is fixed!
The alignment of a noise buffer that is used by the volume view needed to be specified.
Here is the change in the code: [commit link](https://github.com/spectralcode/OCTproZ/commit/0c4a39b5cfc06a5b529c4f68c6714ba7e4072408) 

Please let me know if this fixed the bug for you too.

**Comment by lopmar** [comment]
It works correctly now. Additionally, the 8-bit representation (a02665374766310a0bcb3268fff3785b2b17ec30) was also another issue that you solved quickly after.

Thanks again!
